### ISSUE TITLE
501 5.5.4 Syntax error in parameters or arguments (invalid SIZE parameter)

### ISSUE THREAD

**Issue by baiomys** [axllent/mailpit#407]
Something nasty happening while proxying email via Cloudflare.  
Two different senders, same result:

```
DEBU[2024/12/12 14:35:04] [smtpd] response (104.30.10.33) 220 Greta Mailpit (101) ESMTP Service ready
DEBU[2024/12/12 14:35:04] [smtpd] received (104.30.10.33) EHLO ba-dd.cloudflare-email.net
DEBU[2024/12/12 14:35:04] [smtpd] response (104.30.10.33) 250-Greta greets ba-dd.cloudflare-email.net
250-SIZE 0
250-AUTH LOGIN PLAIN
250 ENHANCEDSTATUSCODES
DEBU[2024/12/12 14:35:04] [smtpd] received (104.30.10.33) MAIL FROM:<SRS0=I7TD=ft=sender.ozon.com=mailer@2tg.org> BODY=8BITMIME
ERRO[2024/12/12 14:35:04] [smtpd] response (104.30.10.33) 501 5.5.4 Syntax error in parameters or arguments (invalid SIZE parameter)

DEBU[2024/12/12 14:35:04] [smtpd] received (104.30.10.33) QUIT

```

**Comment by axllent** [comment]
I think this may be related to to some (currently) unsupported functionality in the smtpd server in Mailpit (I am investigating, however I just had surgery so I'm on very strong pain medication and my brain isn't functioning properly :)).

I'm fairly sure it has to do with the secondary command in `MAIL FROM:<SRS0=I7TD=ft=sender.ozon.com=mailer@2tg.org> BODY=8BITMIME` (ie: ` BODY=8BITMIME`) which is causing the error in Mailpit. I need to do a lot of reading before I can comment further.

In the meantime, can you maybe answer a question: do you get this error if you send directly to Mailpit? I'm curious as to whether it is the CloudFlare proxy that is adding that, or your SMTP client.

**Comment by baiomys** [comment]
> In the meantime, can you maybe answer a question: do you get this error if you send directly to Mailpit?

This weird behavior caused by Cloudflare forwarding in one mailbox to one mailbox rule. 
Catch-all rule (many to one) seems working fine and sending mail directly to mailpit works fine.

**Comment by axllent** [comment]
I believe in some cases SMTP relay servers/proxies (in this case CloudFlare) may be adding the `BODY=...` parameter to assist with message encoding with downstream bounces. Mailpit was originally detecting any parameter as a "size value" which was breaking this. I've pushed a fix for this which is currently in the Docker edge build (unreleased) in case you wanted to test early (if you are able to test please).

Hopefully I will have a new release out by the end of the weekend.

**Comment by baiomys** [comment]
There is some progress evident.
=)
```
mailpit       | time="2024/12/13 06:45:44" level=debug msg="[smtpd] received (192.168.5.3) EHLO ba-cci.cloudflare-email.net"
mailpit       | time="2024/12/13 06:45:44" level=debug msg="[smtpd] response (192.168.5.3) 250-lxc-docker greets ba-cci.cloudflare-email.net\r\n250-SIZE 0\r\n250-AUTH LOGIN PLAIN\r\n250 ENHANCEDSTATUSCODES"
mailpit       | time="2024/12/13 06:45:45" level=debug msg="[smtpd] received (192.168.5.3) MAIL FROM:<SRS0=F0GX=gt=ozon.com=notifications@2tg.org> BODY=8BITMIME"
mailpit       | time="2024/12/13 06:45:45" level=debug msg="[smtpd] response (192.168.5.3) 250 2.1.0 Ok"
mailpit       | time="2024/12/13 06:45:46" level=debug msg="[smtpd] received (192.168.5.3) RCPT TO:<~@gate.2tg.org>"
mailpit       | time="2024/12/13 06:45:46" level=error msg="[smtpd] response (192.168.5.3) 503 5.5.1 Bad sequence of commands (MAIL required before RCPT)"
mailpit       | time="2024/12/13 06:45:46" level=debug msg="[smtpd] received (192.168.5.3) QUIT"
```

**Comment by axllent** [comment]
Damn, that was my mistake, sorry! I have just pushed a fix for the fix. If you don't mind, could you please test again with the latest edge build and let me know if this *now* resolves the issue for you? Thank you!

**Comment by baiomys** [comment]
You fixed it! 
Fantastic, taking into account your current condition. 
I am unable to check status of emails containing SIZE in MAIL FORM string, but I hope it will be also okay.
Thanks for quick fix.

**Comment by axllent** [comment]
This has now been released in [v1.21.7](https://github.com/axllent/mailpit/releases/tag/v1.21.7).
